### PR TITLE
test(x/gamm): add liquidity event tests

### PR DESCRIPTION
**Pull request by p0mvn**
Part of: https://github.com/osmosis-labs/osmosis/issues/1942

## What is the purpose of the change

Tests add liquidity events that should be emitted when a pool is joined.

I noticed that they were emitted twice:
1. at the end of the message server
2. right after a pool state is updated.

I removed 1. while keeping 2. 

## Testing and Verifying

This change added tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/gamm/spec/README.md
Comment (p0mvn) [inline x/gamm/spec/README.md:975]:
```suggestion
This event is emitted after one of `JoinPool` or `JoinPoolNoSwap` complete joining
```

File: x/gamm/spec/README.md
Comment (p0mvn) [inline x/gamm/spec/README.md:982]:
```suggestion
* `sdk.AttributeKeySender`
```

File: x/gamm/spec/README.md
Comment (p0mvn) [inline x/gamm/spec/README.md:984]:
```suggestion
* `types.AttributeKeyPoolId`
```

File: x/gamm/spec/README.md
Comment (p0mvn) [inline x/gamm/spec/README.md:986]:
```suggestion
* `types.AttributeKeyTokensIn`
```

File: x/gamm/keeper/msg_server.go
Comment (mattverse) [inline x/gamm/keeper/msg_server.go:120]: confirmed this event is still happening in `share.go ` in join pool process

File: x/gamm/keeper/internal/events/emit_test.go
Comment (mattverse) [inline x/gamm/keeper/internal/events/emit_test.go:123]: nit: wdyt about moving this variable outside the testcases?

File: x/gamm/keeper/msg_server_test.go
Comment (mattverse) [inline x/gamm/keeper/msg_server_test.go:237]:
```suggestion
		"successful join": {
```

**Review by mattverse — APPROVED** [review]
LGTM! 

Nice catch on removing the existing emit event! 
I do have concern over not being able to intuitively track what events are happening from the msg server layer, rather have to follow inside methods to track what events are happening, but apart from that small concern I have LGTM!

File: x/gamm/keeper/internal/events/emit_test.go
Comment (p0mvn) [reply x/gamm/keeper/internal/events/emit_test.go:123]: I'm not sure how we could do this since expected attributes are dynamically injected into the expected event from the `tc` struct

**Comment by p0mvn** [comment]
> I do have concern over not being able to intuitively track what events are happening from the msg server layer, rather have to follow inside methods to track what events are happening, but apart from that small concern I have LGTM!

I think that's a good point. Let me refactor this at the very end, once I complete all the remaining gamm events. I will add this is an acceptance criteria to the #1942

**Comment by p0mvn** [comment]
I'm merging this since need to unblock further work and the change is not risky

**Pull request merged**
